lisp/ess-r-mode.el

```diff
@@ -2203,6 +2202,71 @@ state.")
 
 
 
+;;;*;;; Inferior R mode
+
+(defvar inferior-ess-r-mode-map
```

Comment:
I don't think this is necessary with the derived mode map. This map is automatically created and is derived from inferior-ess-map.

Reply:
Right, but if we want R specific commands in the inferior we need this. Right now that's just `ess-dirs`, which used to get set by `inferior-ess-mode` with this:

```
  (when (string= ess-language "S") ;; todo: what is this doing here?
    (local-set-key "\M-\r"    'ess-dirs))
```

Reply:
I don't think we strictly need this. One can simply do `(define-key 'inferior-ess-r-mode-map ...)` after `inferior-ess-r-mode` has been defined. Or, if I recall correctly, it could be done within the :map argumen in `define-derived-mode`. 

My main reason to worry about this is that `inferior-ess-r-mode-map` is not derived from `inferior-ess-map` if `defvar` like that, but I might be wrong. `define-derived-mode` might be doing some side-effecting here. Even if it does, it's better if things are explicit.